starlette/requests.py

```diff
@@ -172,10 +172,15 @@ def state(self) -> State:
             self._state = State(self.scope["state"])
         return self._state
 
-    def url_for(self, name: str, **path_params: typing.Any) -> str:
+    def url_for(
+        self, name: str, query_params: dict = None, **path_params: typing.Any
```

Comment:
I am not convinced that we need `query_params` when we return `URL`. It just complicates the signature. 

```python
url = request.url_for('search', query_params={'term': 'findme'})
# vs
url = request.url_for('search').include_query_params(term='findme')
```

Reply:
This MR was just to get the discussion started, now that looks like there's an agreement I will update the PR.